server/container/util-java8/src/main/java/org/apache/james/util/OptionalUtils.java

```diff
@@ -45,4 +45,17 @@
             .filter(Optional::isPresent)
             .orElse(optional2);
     }
+
+    public static <T> boolean containsDifferent(Optional<T> requestValue, T storeValue) {
+        return requestValue
+            .filter(value -> !value.equals(storeValue))
+            .isPresent();
+    }
+
+    public static <T> boolean containsDifferentContent(Optional<T> requestValue, Optional<T> storeValue) {
```

Comment:
Hum ``containsDifferentContent(Optional.empty(), Optional.of(2)) == false`` but ``containsDifferentContent(Optional.of(2), Optional.empty()) == true``

If I am not wrong this is weird

Reply:
good catch but that's our expectation so I changed naming method